daemons/dss-sync/app.py

```diff
@@ -24,7 +29,93 @@ def process_new_syncable_object(event, context):
     if event.get("Event") == "s3:TestEvent":
         app.log.info("DSS sync daemon received S3 test event")
     else:
-        s3 = boto3.resource("s3")
-        bucket = s3.Bucket(event['Records'][0]["s3"]["bucket"]["name"])
+        bucket = resources.s3.Bucket(event['Records'][0]["s3"]["bucket"]["name"])
         obj = bucket.Object(unquote(event['Records'][0]["s3"]["object"]["key"]))
-        sync_blob(source_platform="s3", source_key=obj.key, dest_platform="gs", logger=app.log)
+        sync_blob(source_platform="s3", source_key=obj.key, dest_platform="gs", logger=app.log, context=context)
+
+@app.sns_topic_subscriber("dss-gs-composite-upload-ready")
+def compose_upload(event, context):
+    msg = json.loads(event["Records"][0]["Sns"]["Message"])
+    gs = dss.Config.get_cloud_specific_handles("gcp")[0].gcp_client
+    gs_bucket = gs.get_bucket(msg["dest_bucket"])
+    while True:
+        try:
+            context.log("Composing, stage 1")
+            compose_stage2_blob_names = []
+            if msg["total_parts"] > 32:
+                for part_id in range(1, msg["total_parts"] + 1, 32):
+                    parts_to_compose = range(part_id, min(part_id + 32, msg["total_parts"] + 1))
+                    source_blob_names = ["{}.part{}".format(msg["dest_key"], p) for p in parts_to_compose]
```

Comment:
I have switched to f-style strings wherever possible. But I don't like using f-style strings with complicated constructs and function calls inside. So I still prefer `.format()` in those cases, if that makes sense.